### ISSUE TITLE
Issue with endpoint `editoast/infra/<INFRA_ID>/errors/?`

### ISSUE THREAD

**Issue by bloussou** [OpenRailAssociation/osrd#5089]
### What happened?

If both filters are at `ALL` then response is not paginated.
IF one of the filter is not at all responses are paginated and the size of the response is `25`.

### What did you expect to happen?

Always have a paginated response or not

### How can we reproduce it (as minimally and precisely as possible)?

.

### What operating system, browser and environment are you using?

.

### OSRD version (top right corner `Account` button > `Informations`)

1e051e08

**Comment by leovalais** [comment]
Alright, soooo...

1. The pagination works. It's the content that's huge.
2. Specifically, its the geo and sch information of the `overlapping_catenaries_error` that carry too much data
3. The error type filter also work, but since there is no item "Overlapping catenaries" in the selection list, the only way to make the problem appear is by selecting "All"

It's not the first time that the size of geographic info is problematic. Maybe we should try to find a real fix for that. Off the top of my head, maybe instead of returning the GeoJSON every single time, we could:

* return a ref to the data every time
* expose an endpoint to batch query the data by listing the refs we want to get
* that way, the front could lazily query just the necessary geojsons
* even cache them, maybe

**Comment by leovalais** [comment]
Everywhere a GeoJSON is returned by editoast we could do:

```json
{
    "foo": "bar",
    "geo": {
       "type": "LineString",
       ...
    },
    "sch": { "ref": "<geojson coordinates>" }
}
```

where the geojson coordinates could be a tuple `(<infra_id>, <obj_id>, <geo_or_sch>)` or something more immediate like `(<table>, <row_id>, <geo_or_sch>)`. We could base64 these coordinates to be able to change the format by making them opaque and allow the front to use them as a key for some store or whatever.

Then, we need an endpoint that would behave like:

```http
POST /geojson

["<coord 1>", "<coord 2>", "<coord 3>"]

# response

{
    "<coord 1>": { ... },
    "<coord 2>": { ... },
    "<coord 3>": { ... }
}
```

It would just parse the coordinates, group them by table, issue the SQL request and return the result.

If the front decides to implement a persistent cache of the geojson based on their coordinates, it might be necessary to include a timestamp / invalidation mechanism in order to deal with data being moved in the DB. Tbd I guess...

@flomonster @Tguisnet wdyt?

**Comment by leovalais** [comment]
closed by #5430
